app/cdap/components/Connections/index.tsx

```diff
@@ -41,6 +41,30 @@ const useStyle = makeStyle(() => {
   };
 });
 
+function RedirectIfDefaultConnection({ path, connectionId, initialConnectionId }) {
+  const params = useParams();
+
+  // useRef instead of useState to avoid risk of feedback loop
+  const redirected = useRef(false);
+
+  if (
+    // nothing provided by parent of <Connections/>
+    !connectionId &&
+    // but we have a default
+    initialConnectionId &&
+    // and we're not at the default
+    (params as any).connectionid !== initialConnectionId &&
```

Comment:
If I follow a link like `http://localhost:11011/cdap/ns/default/connections/connection-name-here`, will this redirect to the default?

Reply:
Good call, that should have been !params.connectionid, except it shouldn't have been that as it wasn't in a route that would gate it correctly. See new review.